app/javascript/controllers/handle_theme_controller.js

```diff
@@ -0,0 +1,40 @@
+import { Controller } from '@hotwired/stimulus';
+
+export default class HandleThemeController extends Controller {
+  connect() {
```

Comment:
This connect method is doing quite a lot. We should be able to break this down into smaller single responsibility functions within the class. Anything with the const or function keyword before it is a good candidate for a query or command function.

The [lesson toc controller](https://github.com/TheOdinProject/theodinproject/blob/main/app/javascript/controllers/lesson_toc_controller.js) might be helpful as an example.